src/placeos-frontend-loader/gitremote.cr

```diff
@@ -0,0 +1,197 @@
+      end
+    end
+
+    def github_url
+      "https://github.com/#{@ref}"
+    end
+
+    def branches : Hash(String, String)
+      uri = "https://api.github.com/repos/#{@ref}/branches"
+      response = HTTP::Client.get uri
+      raise Exception.new("status_code for #{uri} was #{response.status_code}") unless response.status_code < 400
+      parsed2 = JSON.parse(response.body).as_a
+      branches = Hash(String, String).new
+      parsed2.each do |value|
+        next if value =~ /HEAD/
+        branch_name = value["name"].to_s.strip.lchop("origin/")
+        branches[branch_name] = value["commit"]["sha"].to_s
+      end
+      branches
+    end
+
+    def tags : Array(String)
+      ["df", "fd"]
+    end
+
+    def download(
+      repository_folder_name : String,
+      content_directory : String,
+      branch : String,
+      repository_commit : String? = nil,
+      username : String? = nil,
+      password : String? = nil
+    )
+      Git.repository_lock(repository_folder_name).write do
+        Log.info { {
+          message:    "downloading repository",
+          repository: repository_folder_name,
+          branch:     branch,
+          uri:        @ref,
+        } }
+      end
+
+      repository_uri = github_url
+
+      begin
+        if branch != "master"
+          hash = get_hash_by_branch(repository_uri, branch)
+        else
+          if repository_commit.nil? || repository_commit == "HEAD"
+            hash = get_hash_head(repository_uri)
+          else
+            hash = repository_commit
+          end
+        end
+        hash = hash.not_nil!
+        tar_url = "#{repository_uri}/archive/#{hash}.tar.gz"
+        download_file(tar_url, TAR_NAME)
+        dest_path = Path.new([content_directory, repository_folder_name])
+        extract_file(TAR_NAME, dest_path)
+        save_metadata(dest_path, hash, repository_uri, branch)
+      rescue ex : Exception
+        Log.error(exception: ex) { ex.message }
+      end
+    end
+
+    def commits(branch : String = @current_branch) : Array(Commit)
+      url = "https://api.github.com/repos/#{@ref}/commits?sha=#{branch}"
+      response = HTTP::Client.get url
+      raise Exception.new("status_code for #{url} was #{response.status_code}") unless response.status_code < 400
+      commits = Array(Commit).new
+      parsed = JSON.parse(response.body).as_a
+      parsed.each do |value|
+        commit = Commit.new(
+          commit: value["sha"].as_s,
+          date: value["commit"]["author"]["date"].as_s,
+          author: value["commit"]["author"]["name"].as_s,
+          subject: value["commit"]["message"].as_s.strip(%(\n))
+        )
+        commits << commit
+      end
+      commits
+    end
+
+    def download_file(url, dest)
+      HTTP::Client.get(url) do |redirect_response|
+        raise Exception.new("status_code for #{url} was #{redirect_response.status_code}") unless redirect_response.status_code < 400
```

Comment:
Since you're not raising any specific exception, you can raise a general exception like this
```suggestion
        raise "status_code for #{url} was #{redirect_response.status_code}" unless redirect_response.success?
```

You can also use `.success?` instead of `< 400` 
I think a gotcha with the stdlib HTTP API is that it doesn't follow redirects, something to keep in mind